### PR TITLE
[core, python] json-monitoring-api-added

### PR DESCRIPTION
**Pull request by rex-schilasky**
### Description
JSON monitoring API added for C++ and Python.

### CODE REVIEW

File: ecal/core/src/monitoring/ecal_monitoring_impl.cpp [hunk 1]
```diff
@@ -553,6 +567,18 @@ namespace eCAL
...
   }
 
+  void CMonitoringImpl::GetMonitoringJSON(std::string& monitoring_, unsigned int entities_)
+  {
+    monitoring_.clear();
+
+    eCAL::pb::Monitoring monitoring_pb;
+    GetMonitoringPb(monitoring_pb, entities_);
+
+    google::protobuf::util::JsonPrintOptions options;
```
Comment (KerstinKeller) [inline ecal/core/src/monitoring/ecal_monitoring_impl.cpp:577]: I think we have some other options which we always set, like to print empty (0) fields. Maybe take a look at the json subscriber?

Comment (rex-schilasky) [reply ecal/core/src/monitoring/ecal_monitoring_impl.cpp:577]:
```
google::protobuf::util::JsonOptions options;
#if GOOGLE_PROTOBUF_VERSION >= 5026000
options.always_print_fields_with_no_presence = true;
#else
options.always_print_primitive_fields = true;
#endif

```

Comment (KerstinKeller) [reply ecal/core/src/monitoring/ecal_monitoring_impl.cpp:577]: jo

File: lang/python/core/src/ecal_wrap.cxx [hunk 2]
```diff
@@ -1525,6 +1525,16 @@ PyObject* mon_monitoring(PyObject* /*self*/, PyObject* /*args*/)
...
 }
 
+/****************************************/
+/*      mon_monitoring_json             */
+/****************************************/
+PyObject* mon_monitoring_json(PyObject* /*self*/, PyObject* /*args*/)
+{
+  std::string monitoring_s;
+  eCAL::Monitoring::GetMonitoringJSON(monitoring_s);
+  return(Py_BuildValue("s", ecal_getversion()));
```
Comment (KerstinKeller) [inline lang/python/core/src/ecal_wrap.cxx:1535]: this is never, ever correct 😄

Comment (rex-schilasky) [reply lang/python/core/src/ecal_wrap.cxx:1535]: it's not :-/

Comment (rex-schilasky) [reply lang/python/core/src/ecal_wrap.cxx]: fixed

**Comment by rex-schilasky** [comment]
we can not merge this without API break

**Pull request closed without merging**